src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/ExternalProcessRunner.java

```diff
@@ -132,15 +136,25 @@ public ProcessResult executeAndThrowOnFailure(String command, String... args) th
         ProcessResult processResult = execute(command, args);
         // if the process fails we attach the output into the exception
         if (processResult.getExitCode() != 0) {
-            String lineSep = System.lineSeparator();
-            String processOutputString = String.join(lineSep, processResult.processOutput);
+            String processOutputString = processOutputToString(processResult.processOutput);
             String errorMsg = String.format("Failed to run command '%s' with output: %n %s",
                     processResult.commandWithArgs, processOutputString);
+            log.error(errorMsg);
             throw new ProcessFailedException(errorMsg);
         }
         return processResult;
     }
 
+    /**
+     * Format the process output string list to one string
+     * @param processOutput
+     * @return
+     */
+    public static String processOutputToString(List<String> processOutput) {
```

Comment:
This formats the string so that it does have proper line breaks. It is meant for logging purposes and does not affect on the output itself.

However I did some digging and noticed that when validating a WSDL, the output is not attached to the exception metadata at all (always leaving metadata empty even though the WSDL validation has some complaints.) I updated the [executeAndThrowOnFailure](https://github.com/nordic-institute/X-Road/pull/450/commits/97d377c7c058c7785f099719a7fa383d08eb0236#diff-233397d530bd3c9726636073bf676977R143) so that the process output is attached to the exception metadata.